File: dev-docs/snippets/documents/create-invitation.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Deprecated: prefer `updatePermissions({ email, permission })`.
// Create a legacy per-document invitation by email.
export async function createInvitation(client: JsBaoClient, documentId: string) {
  // #region example
  const result = await client.documents.createInvitation(
    documentId,
    "teammate@example.com",
    "read-write",
    { sendEmail: true },
  );
  // #endregion example
  return result;
}

File: dev-docs/snippets/documents/create-invitation.swift
import JsBaoClient

// Deprecated: prefer `updatePermissions(params:)` with `.email(...)`. In Swift
// this method is named `sendInvitation` and returns a typed
// `DocumentInvitationResponse`.
func createInvitation(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let result = try await client.documents.sendInvitation(
    documentId: documentId,
    email: "teammate@example.com",
    permission: "read-write",
    options: InvitationEmailOptions(sendEmail: true)
  )
  // #endregion example
  _ = result.invitationId
}
